### ISSUE TITLE
Support for multiclass sentence classification

### ISSUE THREAD

**Issue by paradc2** [google-research/bert#200]
I am encountering an issue training multiclass sentence classifiers. While first encountered training a custom data processor, I can replicate it using the mnli data processor out of the box. Steps to reproduce: 

Execute run_classifier.py using the shell script provided in the [sentence classifier guide](https://github.com/google-research/bert#sentence-and-sentence-pair-classification-tasks) with the task and data directory set to mnli. The following error is raised:

> InvalidArgumentError (see above for traceback): Restoring from checkpoint failed. This is most likely due to a mismatch between the current graph and the graph from the checkpoint. Please ensure that you have not altered the graph expected based on the checkpoint. Original error:
> 
> Assign requires shapes of both tensors to match. lhs shape= [3,768] rhs shape= [2,768]
> 	 [[{{node save/Assign_602}} = Assign[T=DT_FLOAT, _class=["loc:@output_weights/adam_v"], use_locking=true, validate_shape=true, _device="/job:localhost/replica:0/task:0/device:CPU:0"](output_weights/adam_v, save/RestoreV2:603)]]

It seems the BERT checkpoint is expecting only two classes. I am using a fresh download BERT-Base, Uncased, with Python 2.7.13 and Tensorflow 1.11.0. Is there a step I am missing for multiclass problems, or is it a separate issue? 

Full stack trace:

> Caused by op u'save/Assign_602', defined at:
>   File "run_classifier.py", line 912, in <module>
>     tf.app.run()
>   File "/Users/cparadis/anaconda/envs/aml_main/lib/python2.7/site-packages/tensorflow/python/platform/app.py", line 125, in run
>     _sys.exit(main(argv))
>   File "run_classifier.py", line 835, in main
>     estimator.train(input_fn=train_input_fn, max_steps=num_train_steps)
>   File "/Users/cparadis/anaconda/envs/aml_main/lib/python2.7/site-packages/tensorflow/contrib/tpu/python/tpu/tpu_estimator.py", line 2394, in train
>     saving_listeners=saving_listeners
>   File "/Users/cparadis/anaconda/envs/aml_main/lib/python2.7/site-packages/tensorflow/python/estimator/estimator.py", line 356, in train
>     loss = self._train_model(input_fn, hooks, saving_listeners)
>   File "/Users/cparadis/anaconda/envs/aml_main/lib/python2.7/site-packages/tensorflow/python/estimator/estimator.py", line 1181, in _train_model
>     return self._train_model_default(input_fn, hooks, saving_listeners)
>   File "/Users/cparadis/anaconda/envs/aml_main/lib/python2.7/site-packages/tensorflow/python/estimator/estimator.py", line 1215, in _train_model_default
>     saving_listeners)
>   File "/Users/cparadis/anaconda/envs/aml_main/lib/python2.7/site-packages/tensorflow/python/estimator/estimator.py", line 1406, in _train_with_estimator_spec
>     log_step_count_steps=self._config.log_step_count_steps) as mon_sess:
>   File "/Users/cparadis/anaconda/envs/aml_main/lib/python2.7/site-packages/tensorflow/python/training/monitored_session.py", line 504, in MonitoredTrainingSession
>     stop_grace_period_secs=stop_grace_period_secs)
>   File "/Users/cparadis/anaconda/envs/aml_main/lib/python2.7/site-packages/tensorflow/python/training/monitored_session.py", line 921, in __init__
>     stop_grace_period_secs=stop_grace_period_secs)
>   File "/Users/cparadis/anaconda/envs/aml_main/lib/python2.7/site-packages/tensorflow/python/training/monitored_session.py", line 643, in __init__
>     self._sess = _RecoverableSession(self._coordinated_creator)
>   File "/Users/cparadis/anaconda/envs/aml_main/lib/python2.7/site-packages/tensorflow/python/training/monitored_session.py", line 1107, in __init__
>     _WrappedSession.__init__(self, self._create_session())
>   File "/Users/cparadis/anaconda/envs/aml_main/lib/python2.7/site-packages/tensorflow/python/training/monitored_session.py", line 1112, in _create_session
>     return self._sess_creator.create_session()
>   File "/Users/cparadis/anaconda/envs/aml_main/lib/python2.7/site-packages/tensorflow/python/training/monitored_session.py", line 800, in create_session
>     self.tf_sess = self._session_creator.create_session()
>   File "/Users/cparadis/anaconda/envs/aml_main/lib/python2.7/site-packages/tensorflow/python/training/monitored_session.py", line 557, in create_session
>     self._scaffold.finalize()
>   File "/Users/cparadis/anaconda/envs/aml_main/lib/python2.7/site-packages/tensorflow/python/training/monitored_session.py", line 215, in finalize
>     self._saver.build()
>   File "/Users/cparadis/anaconda/envs/aml_main/lib/python2.7/site-packages/tensorflow/python/training/saver.py", line 1106, in build
>     self._build(self._filename, build_save=True, build_restore=True)
>   File "/Users/cparadis/anaconda/envs/aml_main/lib/python2.7/site-packages/tensorflow/python/training/saver.py", line 1143, in _build
>     build_save=build_save, build_restore=build_restore)
>   File "/Users/cparadis/anaconda/envs/aml_main/lib/python2.7/site-packages/tensorflow/python/training/saver.py", line 781, in _build_internal
>     restore_sequentially, reshape)
>   File "/Users/cparadis/anaconda/envs/aml_main/lib/python2.7/site-packages/tensorflow/python/training/saver.py", line 459, in _AddShardedRestoreOps
>     name="restore_shard"))
>   File "/Users/cparadis/anaconda/envs/aml_main/lib/python2.7/site-packages/tensorflow/python/training/saver.py", line 428, in _AddRestoreOps
>     assign_ops.append(saveable.restore(saveable_tensors, shapes))
>   File "/Users/cparadis/anaconda/envs/aml_main/lib/python2.7/site-packages/tensorflow/python/training/saver.py", line 119, in restore
>     self.op.get_shape().is_fully_defined())
>   File "/Users/cparadis/anaconda/envs/aml_main/lib/python2.7/site-packages/tensorflow/python/ops/state_ops.py", line 221, in assign
>     validate_shape=validate_shape)
>   File "/Users/cparadis/anaconda/envs/aml_main/lib/python2.7/site-packages/tensorflow/python/ops/gen_state_ops.py", line 61, in assign
>     use_locking=use_locking, name=name)
>   File "/Users/cparadis/anaconda/envs/aml_main/lib/python2.7/site-packages/tensorflow/python/framework/op_def_library.py", line 787, in _apply_op_helper
>     op_def=op_def)
>   File "/Users/cparadis/anaconda/envs/aml_main/lib/python2.7/site-packages/tensorflow/python/util/deprecation.py", line 488, in new_func
>     return func(*args, **kwargs)
>   File "/Users/cparadis/anaconda/envs/aml_main/lib/python2.7/site-packages/tensorflow/python/framework/ops.py", line 3272, in create_op
>     op_def=op_def)
>   File "/Users/cparadis/anaconda/envs/aml_main/lib/python2.7/site-packages/tensorflow/python/framework/ops.py", line 1768, in __init__
>     self._traceback = tf_stack.extract_stack()

**Comment by prathyushaks** [comment]
+1 I have the same issue when I try to run this on pytorch. For some reason for the MNLI dataset, the code provides a class error since the code is expecting binary classes. Is there a way to test on the SST-2 data?SST-2 is binary.

**Comment by kyrocera** [comment]
Enjoy the bug & learn from it. Replicate it. You'll debug before you even
recreate it

**Comment by kyrocera** [comment]
Lol nerds

Nerf em
